src/helpers/reddit_client.ts

```diff
@@ -56,4 +75,56 @@ export class RedditClient {
             return [];
         }
     }
+
+    async getTopDayPosts(): Promise<Array<KpopNewsRedditPost>> {
+        try {
+            const matchingPosts = await this.client.search({
+                subreddit: "kpop",
+                query: generateFilteredQuery(),
+                sort: "top",
+                time: "day",
+            });
+
+            const popularPosts = matchingPosts.filter((x) => x.score > 100);
+
+            return popularPosts.map((x) => ({
+                title: x.title,
+                link: `https://reddit.com${x.permalink}`,
+                date: new Date(x.created_utc * 1000),
+                flair: x.link_flair_css_class as string,
+                x: Date.now() - new Date(x.created_utc * 1000).getTime(),
+            }));
+        } catch (e) {
+            logger.error(
+                `Failed to fetch getTopDayPosts(). e = ${JSON.stringify(e)}`,
+            );
+            return [];
+        }
+    }
+
+    async getTopWeekPosts(): Promise<Array<KpopNewsRedditPost>> {
```

Comment:
could we share logic between daily/weekly/monthly somehow? most of it seems teh same